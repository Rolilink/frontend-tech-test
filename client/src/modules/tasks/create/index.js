import connectToState from './providers/CreateTask';
import CreateTaskComponent from './components/CreateTask';

export default connectToState(CreateTaskComponent);
