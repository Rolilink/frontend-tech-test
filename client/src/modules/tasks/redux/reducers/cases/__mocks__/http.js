export const setSuccess = jest.fn(() => ({}));
export const setLoading = jest.fn(() => ({}));
export const setError = jest.fn(() => ({}));
