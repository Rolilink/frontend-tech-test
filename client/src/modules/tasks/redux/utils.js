export default (tasks) => {
  const normalizedTasks = {};

  tasks.forEach((task) => {
    normalizedTasks[task.id] = task;
  });

  return normalizedTasks;
};
