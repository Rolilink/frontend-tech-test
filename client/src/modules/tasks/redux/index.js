import * as actionTypes from './actionTypes';
import * as factories from './factories';
import * as selectors from './selectors';
import * as utils from './utils';

import { fetchTasks, createTask } from './actions';

import reducerModule from './reducers';

const reducer = {
  name: 'tasks', reducer: reducerModule,
};

export {
  fetchTasks,
  createTask,
  actionTypes,
  factories,
  selectors,
  reducer,
  utils,
};
