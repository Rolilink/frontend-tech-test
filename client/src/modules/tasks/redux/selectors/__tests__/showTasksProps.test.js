import { taskFactory } from '../../../../../__tests__/testUtils/tasks';
import { httpFactory } from '../../../../../__tests__/testUtils/http';
import { normalizeTasks } from '../../utils';
import getShowTasksProps from '../showTasksProps';

const state = {
  tasks: {
    allTasks: {
      tasks: {
        ...normalizeTasks([
          taskFactory({ id: 1 }),
          taskFactory({ id: 2 }),
          taskFactory({ id: 3 }),
        ]),
      },
      ...httpFactory(
        {
          error: 'error',
          isFetching: true,
        },
      ),
    },
  },
};


describe('Tasks.Redux.Selectors.getShowTasksProps', () => {
  it('should get isFetching from allTasks when receiving a state', () => {
    expect(getShowTasksProps(state).isFetching).toBe(true);
  });

  it('should get error from allTasks when receiving a state', () => {
    expect(getShowTasksProps(state).error).toBe('error');
  });

  it('should get visible tasks from allTasks when receiving a state', () => {
    expect(getShowTasksProps(state).tasks).toMatchSnapshot();
  });
});
