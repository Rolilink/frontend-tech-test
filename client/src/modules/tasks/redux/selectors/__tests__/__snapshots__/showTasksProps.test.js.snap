// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks.Redux.Selectors.getShowTasksProps should get visible tasks from allTasks when receiving a state 1`] = `
Array [
  Object {
    "description": "Description",
    "id": 1,
    "title": "Title",
  },
  Object {
    "description": "Description",
    "id": 2,
    "title": "Title",
  },
  Object {
    "description": "Description",
    "id": 3,
    "title": "Title",
  },
]
`;
